### ISSUE TITLE
Heating coil capacity specified in Sizing:System does not flow into Coil:Heating:Water capacity correctly

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#5803]
## Issue overview

_From user_
using Sizing:System to assign heating and cooling capacities to VAV system 'AHU.WP' in my IDF. The cooling coil size reported in the output table matches my sizing input in 'cooling design capacity input' (139,000 W). The reported heating coil sizing (67,574.1 W) however, does not match my 'heating design capacity input' of 46,000 W. I tried a number of different capacities, but the reported autosize value is generally unaffected. 

_Reply_
With 46000 in Sizing:System

```
 Component Sizing Information, Coil:Heating:Water, AHU.WP HEATING COIL, Design Size Design Coil Load [W], 46000.00000
 Component Sizing Information, Coil:Heating:Water, AHU.WP HEATING COIL, Design Size Maximum Water Flow Rate [m3/s], 9.99342E-004
 Component Sizing Information, Coil:Heating:Water, AHU.WP HEATING COIL, Design Size U-Factor Times Area Value [W/K], 1565.90514
Water Heating Coil Capacity Information,Coil:Heating:Water,AHU.WP HEATING COIL,68126.11
```

With 40000 in Sizing:System

```
 Component Sizing Information, Coil:Heating:Water, AHU.WP HEATING COIL, Design Size Design Coil Load [W], 40000.00000
 Component Sizing Information, Coil:Heating:Water, AHU.WP HEATING COIL, Design Size Maximum Water Flow Rate [m3/s], 8.68993E-004
 Component Sizing Information, Coil:Heating:Water, AHU.WP HEATING COIL, Design Size U-Factor Times Area Value [W/K], 1626.82016
Water Heating Coil Capacity Information,Coil:Heating:Water,AHU.WP HEATING COIL,68126.11
```

So, the "reported" max water flow rate and UA change, but the reported nominal capacity is _exactly_ the same.

With 40000 in the Coil:Heating:Water object Nominal Capacity field, the reported nominal capacity is nearly 40000, and the water flow rate is yet another value.

```
Water Heating Coil Capacity Information,Coil:Heating:Water,AHU.WP HEATING COIL,39985.52
 Component Sizing Information, Controller:WaterCoil, AHU.WP HEATING COIL CONTROLLER, Maximum Actuated Flow [m3/s], 8.61164E-004
```
### Details

Some additional details for this issue (if relevant):
- Platform (Win64)
- Version of EnergyPlus (v8.6.0-53f5fc33af)
- Helpdesk ticket number 11504
- AUS_VIC.Melbourne.948680_RMY.epw
### Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [x] Defect file added (list location of defect file here)
- [ ] Ticket added to Pivotal for defect (development team task)
- [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by wfbuhl** [comment]
There's something more (or different) going on here. Here's the results for the 5 reheat coils in 5ZoneWarmest:

Water Heating Coil Capacity Information,Coil:Heating:Water,SPACE1-1 ZONE COIL,9996.61
Water Heating Coil Capacity Information,Coil:Heating:Water,SPACE2-1 ZONE COIL,9996.61
Water Heating Coil Capacity Information,Coil:Heating:Water,SPACE3-1 ZONE COIL,9996.61
Water Heating Coil Capacity Information,Coil:Heating:Water,SPACE4-1 ZONE COIL,9996.61
Water Heating Coil Capacity Information,Coil:Heating:Water,SPACE5-1 ZONE COIL,9996.61

Hmm. Admittedly all 5 coils have the same entering air temperature, entering humidity ratio, entering water temperature because this is a rated capacity. But they have different air and water flow rates. Capacity is calculated by firing the heating coil calculation routine.

This all happens near the end of InitWaterCoil. I know I'll regret saying this, but this can't be too hard to fix.

**Comment by wfbuhl** [comment]
There are a couple of minor things, and one big fix. First the minor stuff.
1. Error in IORef doc: NominalCapacity is not the default for Coil:Heating:Water. It is UFactorTimesAreaAndDesignWaterFlowRate. This is in "Field: Performance Input Method".
2. There's an error in 1.17.8.1.26 (in Sizing:System): it says FlowPerCoolingCapacity instead of FlowPerHeatingCapacity.
3. Here is the user's input for Sizing:System:
   
   Sizing:System,
     AHU.WP,                  !- AirLoop Name
     Total,                   !- Type of Load to Size On
     0.9,                     !- Design Outdoor Air Flow Rate {m3/s}
     0.3,                     !- Central Heating Maximum System Air Flow Ratio
     7.2,                     !- Preheat Design Temperature {C}
     0.008,                   !- Preheat Design Humidity Ratio {kgWater/kgDryAir}
     11,                      !- Precool Design Temperature {C}
     0.008,                   !- Precool Design Humidity Ratio {kgWater/kgDryAir}
     12.8,                    !- Central Cooling Design Supply Air Temperature {C}
     35,                      !- Central Heating Design Supply Air Temperature {C}
     NonCoincident,           !- Type of Zone Sum to Use
     No,                      !- 100% Outdoor Air in Cooling
     No,                      !- 100% Outdoor Air in Heating
     0.008,                   !- Central Cooling Design Supply Air Humidity Ratio {kgWater/kgDryAir}
     0.008,                   !- Central Heating Design Supply Air Humidity Ratio {kgWater/kgDryAir}
     FlowPerCoolingCapacity,  !- Cooling Supply Air Flow Rate Method
     0,                       !- Cooling Supply Air Flow Rate {m3/s}
     ,                        !- Cooling Supply Air Flow Rate Per Floor Area {m3/s-m2}
     ,                        !- Cooling Fraction of Autosized Cooling Supply Air Flow Rate
     0.0000601671,            !- Cooling Supply Air Flow Rate Per Unit Cooling Capacity {m3/s-W}
     FlowPerHeatingCapacity,  !- Heating Supply Air Flow Rate Method
     0,                       !- Heating Supply Air Flow Rate {m3/s}
     ,                        !- Heating Supply Air Flow Rate Per Floor Area {m3/s-m2}
     ,                        !- Heating Fraction of Autosized Heating Supply Air Flow Rate
     ,                        !- Heating Fraction of Autosized Cooling Supply Air Flow Rate
     0.000174194,             !- Heating Supply Air Flow Rate Per Unit Heating Capacity {m3/s-W}
     ZoneSum,                 !- System Outdoor Air Method
     1,                       !- Zone Maximum Outdoor Air Fraction {dimensionless}
     CoolingDesignCapacity,   !- Cooling Design Capacity Method
     139000,                  !- Cooling Design Capacity {W}
     ,                        !- Cooling Design Capacity Per Floor Area {W/m2}
     ,                        !- Fraction of Autosized Cooling Design Capacity
     HeatingDesignCapacity,   !- Heating Design Capacity Method
     46000,                   !- Heating Design Capacity {W}
     ,                        !- Heating Design Capacity Per Floor Area {W/m2}
     ,                        !- Fraction of Autosized Heating Design Capacity
     VAV;                     !- Central Cooling Capacity Control Method

Heating Supply Air Flow Rate Method = FlowPerHeatingCapacity: this flow is calculated from the autosized heating capacity not from the capacity entered in  "Heating Design Capacity". The same goes for cooling. This is probably not what the user expects, but it is sort of indicated in the IORef. Also these two inputs can easily be mismatched, especially when the program is not using the design load the user thinks it is.
1. Finally the main issue. Here is the user's water coil input:
   
   Coil:Heating:Water,
     AHU.WP Heating Coil,     !- Name
     ,                        !- Availability Schedule Name
     Autosize,                !- U-Factor Times Area Value {W/K}
     Autosize,                !- Maximum Water Flow Rate {m3/s}
     AHU.WP Heating Coil HW Inlet,  !- Water Inlet Node Name
     AHU.WP Heating Coil HW Outlet,  !- Water Outlet Node Name
     AHU.WP Cooling Coil Outlet,  !- Air Inlet Node Name
     AHU.WP Heating Coil Outlet,  !- Air Outlet Node Name
     NominalCapacity,         !- Performance Input Method
     Autosize,                !- Rated Capacity {W}
     82.2,                    !- Rated Inlet Water Temperature {C}
     16.6,                    !- Rated Inlet Air Temperature {C}
     71.1,                    !- Rated Outlet Water Temperature {C}
     32.2,                    !- Rated Outlet Air Temperature {C}
     0.5;                     !- Rated Ratio for Air and Water Convection

For Performance Input Method = NominalCaoacity the code recalculates the design water mass flow rate and UA in InitWaterCoil. It does not report these values anywhere and uses them in all further calculations. If the rated capacity is autosized it uses the sizing calculation value, not the value the user input back in System:Sizing. The values for design for design water mass flow rate and UA calculated back in SizeWaterCoil do pick up and use the user's input from Sizing:System correctly.

So we need to:

a) eliminate the recalc in InitWaterCoil.

b) move it to SizeWaterCoil (this will be a little messy).

c) Decide what to do if Performance Input Method = NominalCapacity and Rated Capacity = autosize. Should it ignore the rated temperatures, since using them wouldn't make much sense. What happens if this field is blank? Should the rated temperatures only be used if the user input a value > 0 for Rated Capacity?

Meanwhile I will implement a) and b) if it is OK with @mjwitte.

**Comment by mjwitte** [comment]
@wfbuhl a) and b) sound like the right thing to do.  

c) Has been a source of confusion.  Also see #4223.  There it's also suggested that the rated temps be "autosizable", i.e. grabbing the entering and leaving temps from the sizing data.

> Heating Supply Air Flow Rate Method = FlowPerHeatingCapacity: this flow is calculated from the autosized heating capacity not from the capacity entered in "Heating Design Capacity". The same goes for cooling. This is probably not what the user expects, but it is sort of indicated in the IORef. 

I see that now in the I/O Ref, but why?  Seems it should use whatever capacity comes out of Heating Design Capacity Method.  This explains what is happening in #5835 .

**Comment by wfbuhl** [comment]
"I see that now in the I/O Ref, but why? Seems it should use whatever capacity comes out of Heating Design Capacity Method. This explains what is happening in #5835 ."

There's a coding issue if I remember right. The flow is calculated first and it doesn't know about the "Heating Design Capacity" yet. Tho I suppose we could check that first. Maybe has to do with the code development sequence. It is hard to keep track of everything with the multiple methods of input.

**Comment by wfbuhl** [comment]
Oh yes. In SizeWaterCoil, the max water flow autosize does use the Heating Design Capacity. But the UA autosize DOES NOT. For pete's sake. Well, we'll fix this too.

**Comment by mjwitte** [comment]
@wfbuhl Can you address #5821 as part of this fix?

**Comment by wfbuhl** [comment]
Yes

**Comment by Myoldmopar** [comment]
Closed via #5851
